src/tests/end-to-end/common/page-controllers/target-page.ts

```diff
@@ -48,9 +48,9 @@ export class TargetPage extends Page {
     }
 
     public async waitForShadowRoot(): Promise<ElementHandle<Element>> {
-        return await this.waitForSelector('#insights-shadow-host #insights-shadow-container', {
+        return (await this.waitForSelector('#insights-shadow-host #insights-shadow-container', {
             state: 'attached',
-        });
+        }))!;
```

Comment:
This is the most interesting fix. The contract for Playwright's `waitForSelector` is that it will throw rather than returning null if it can't find an element, *except* that if you explicitly pass options like `{state: 'detached'}`, then it will return null in the success path. So `waitForSelectorInShadowRoot` is allowed to return null (since someone could pass through that option), but `waitForShadowRoot` gets to assume that `waitForSelector` is not returning null (since it's passing `state: 'attached'` always)